### PR TITLE
Added Transmodel TRIP REQUEST.Flexibilitywindow

### PR DESCRIPTION
**Pull request by ue71603**

### CODE REVIEW

**Comment by normanoffel** [comment]
What is the use case for this field? When should it be filled and what are its implications / what are the expectations from the enquirers?

**Review by herlitze — APPROVED** [review]

File: OJP/OJP_JourneySupport.xsd [hunk 1]
```diff
@@ -57,6 +57,23 @@
...
 		</xs:sequence>
 	</xs:complexType>
+	<xs:complexType name="DepArrTimeRangeStructure">
+		<xs:annotation>
+			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed thatthe implementation delivers all possible arrivals and departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="DepArrTimeRangeBegin" type="xs:dateTime" minOccurs="1">
+				<xs:annotation>
+					<xs:documentation>Begin of the interval.</xs:documentation>
```
Comment (herlitze) [inline OJP/OJP_JourneySupport.xsd:67]: interval -> time range

File: OJP/OJP_JourneySupport.xsd [hunk 2]
```diff
@@ -57,6 +57,23 @@
...
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="DepArrTimeRangeEnd" type="xs:dateTime" minOccurs="1">
+				<xs:annotation>
+					<xs:documentation>End of the interval.</xs:documentation>
```
Comment (herlitze) [inline OJP/OJP_JourneySupport.xsd:72]: interval -> time range

File: OJP/OJP_JourneySupport.xsd [hunk 3]
```diff
@@ -587,6 +604,11 @@
 						<xs:documentation>Extra time needed before reaching/after leaving this location (an example of a TRIP ACCESS CONSTRAINT. In Transmodel it is modeled more extensively and could relate to ACCESS MODE e.g. walk max 5 mintues, but cycle 10 minutes).</xs:documentation>
 					</xs:annotation>
 				</xs:element>
+				<xs:element name="DepArrTimeRange" type="DepArrTimeRangeStructure" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed thatthe implementation delivers all possible arrivals and departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
```
Comment (herlitze) [inline OJP/OJP_JourneySupport.xsd:609]: - space missing in the third sentence that_the
- i would expect wither arrivals or departures: ... all possible arrivals **or** departures in the interval

**Review by herlitze — DISMISSED** [review]

File: OJP/OJP_JourneySupport.xsd (hunk 2 above)
Comment (ue71603) [inline OJP/OJP_JourneySupport.xsd:72]:
```suggestion
					<xs:documentation>End of the time range.</xs:documentation>
```

File: OJP/OJP_JourneySupport.xsd (hunk 3 above)
Comment (ue71603) [inline OJP/OJP_JourneySupport.xsd:609]:
```suggestion
						<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed that the implementation delivers all possible arrivals or departures in the time range. E.g. for airplane with +/- 3 days.</xs:documentation>
```

File: OJP/OJP_JourneySupport.xsd [hunk 4]
```diff
@@ -57,6 +57,23 @@
 			</xs:element>
 		</xs:sequence>
 	</xs:complexType>
+	<xs:complexType name="DepArrTimeRangeStructure">
+		<xs:annotation>
+			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed thatthe implementation delivers all possible arrivals and departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_JourneySupport.xsd:62]:
```suggestion
			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed that the implementation delivers all possible arrivals and departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
```

**Review by skinkie — APPROVED** [review]

**Review by skinkie — DISMISSED** [review]

**Review by AndreasAtSBB — DISMISSED** [review]
added new TRIP REQUEST.FlexibilityWindow with an example and also new matching error definitions.

File: OJP/OJP_JourneySupport.xsd [hunk 5]
```diff
@@ -57,6 +57,23 @@
 			</xs:element>
 		</xs:sequence>
 	</xs:complexType>
+	<xs:complexType name="DepArrTimeRangeStructure">
+		<xs:annotation>
+			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed that the implementation delivers all possible arrivals and departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
```
Comment (trurlurl) [inline OJP/OJP_JourneySupport.xsd:62]:
```suggestion
			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed that the implementation delivers all possible arrivals or departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
```

**Review by trurlurl — CHANGES_REQUESTED** [review]
1) *PlaceContext* is also used in *StopEventRequest* which already contains the element *TimeWindow*. Isn't that too redundant / confusing?

2) As for the logic how to integrate *DepArrTimeRange* and *NumberOfResults** - I suppose that they are treated to indicate what the results should cover at least (i.e., each one is able to extend the number of results).

File: OJP/OJP_JourneySupport.xsd [hunk 6]
```diff
@@ -57,6 +57,23 @@
 			</xs:element>
 		</xs:sequence>
 	</xs:complexType>
+	<xs:complexType name="DepArrTimeRangeStructure">
+		<xs:annotation>
+			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed that the implementation delivers all possible arrivals or departures in the interval. E.g. for airplane with +/- 3 days.</xs:documentation>
```
Comment (ue71603) [inline OJP/OJP_JourneySupport.xsd:62]:
```suggestion
			<xs:documentation>The acceptable time range for the arrival or departure. Implements TRIP REQUEST.FlexibilityWindow. It is assumed that the implementation delivers all possible arrivals or departures in the time range. E.g. for airplane with +/- 3 days.</xs:documentation>
```

**Review by skinkie — APPROVED** [review]

File: OJP/OJP_JourneySupport.xsd [hunk 7]
```diff
@@ -57,6 +57,23 @@
...
+			<xs:element name="DepArrTimeRangeBegin" type="xs:dateTime" minOccurs="1">
+				<xs:annotation>
+					<xs:documentation>Begin of the time range.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="DepArrTimeRangeEnd" type="xs:dateTime" minOccurs="1">
+				<xs:annotation>
+					<xs:documentation>End of the time range.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
```
Comment (normanoffel) [inline OJP/OJP_JourneySupport.xsd:74]: I would prefer a duration over a end date time specification.

**Review by skinkie — APPROVED** [review]

**Review by trurlurl — APPROVED** [review]

**Review by sgrossberndt — CHANGES_REQUESTED** [review]
On it

**Review by sgrossberndt — APPROVED** [review]
Two improvements:
* [Improve wording and fix typo in documentation](https://github.com/VDVde/OJP/pull/372/commits/ba5dff8d86640b37d5a2801c92edabb639796377)
* [Renamed problem code to TRIP_TIMEWINDOWTOOLARGE](https://github.com/VDVde/OJP/pull/372/commits/0b1328fd1ac09a69cd94eac99d10831058f2cd04) (should not contain "ERROR", will add a pull request for `TRIP_DATETIMEERROR` shortly) and since the problem is about the time window let's use that in the name of the problem as well.

**Review by skinkie — APPROVED** [review]

**Pull request merged**